shouldupdate.js

```diff
@@ -147,10 +106,17 @@ function factory (methods) {
    */
   function isEqualState (value, other) {
     return isEqual(value, other, function (current, next) {
-      if (_isImmutable(current) && _isImmutable(next)) {
-        return current === next;
+      if (current == next) {
+        return true;
       }
-      if (_isImmutable(current) || _isImmutable(next)) {
+
+      var isCurrentImmutable = _isImmutable(current);
+      var isNextImmutable = _isImmutable(next);
+
```

Comment:
While I kind of like the idea of an interface for saying you want component args to trigger renders or not when changed, I'm still not conviced Immutable.js .equals() is the way to go.